### DISCUSSION TITLE
Dynamic Library Loading and Intune SDK Compatibility Challenges

### DISCUSSION THREAD

**Discussion by vinaykumar0339** [microsoftconnect/ms-intune-app-sdk-ios#500] [General]
**Problem Statement:**
Whenever we need to update the Intune SDK, there are mandatory minimum version checks that require updates to iOS versions (e.g., iOS 14, 16) for major releases. Each time we update the library, we face the risk of losing iOS users due to these version constraints. Additionally, we are restricted to using specific Xcode versions, such as Xcode 14 or Xcode 16.

**Suggestion Needed:**
While increasing the Xcode version is manageable, is there a way to dynamically load the library without needing to increment the minimum iOS version each time? The goal is to continue using the Intune SDK without impacting app usage or introducing compatibility issues

**Comment by pmod2** [comment]
If your goal is to dynamically use different SDK versions in the same app, this is not supported as of now.
